src/twin/system_info.rs

```diff
@@ -58,21 +118,39 @@ impl Feature for SystemInfo {
     }
 
     fn event_stream(&mut self) -> EventStreamResult {
-        if self.boot_time.is_none() {
-            Ok(Some(file_created_stream::<SystemInfo>(vec![
-                &TIMESYNC_FILE,
-            ])))
+        if 0 == *REFRESH_SYSTEM_INFO_INTERVAL_SECS {
+            if self.boot_time.is_none() {
+                Ok(Some(file_created_stream::<SystemInfo>(vec![
+                    &TIMESYNC_FILE,
+                ])))
+            } else {
+                Ok(None)
+            }
         } else {
-            Ok(None)
+            let stream = interval_stream::<SystemInfo>(interval(Duration::from_secs(
+                *REFRESH_SYSTEM_INFO_INTERVAL_SECS,
+            )));
+            Ok(Some(match self.boot_time {
+                None => futures::stream::select(
+                    stream,
+                    file_created_stream::<SystemInfo>(vec![&TIMESYNC_FILE]),
+                )
+                .boxed(),
+                _ => stream,
+            }))
```

Comment:
Merging both streams is absolutely what I had in mind in case we need multiple streams.
The only issue with this block is about readability. Imho you can improve a lot with pattern matching or using `Option` methods. Have a look at `match` docu. E.g. you can make use of `match` plus `if` clauses.

Reply:
refactored, but not sure if it sufficient.

Reply:
fine for me as well

Reply:
done